src/vl-map-wms-layer.js

```diff
@@ -0,0 +1,96 @@
+import {VlMapLayer} from '/src/vl-map-layer.js';
+import {
+  OlImageLayer,
+  OlImageWMSSource,
+  OlTileWMSSource,
+  OlTileLayer,
+} from '/node_modules/vl-mapactions/dist/vl-mapactions.js';
+
+/**
+ * VlMapWmsLayer
+ * @class
+ * @classdesc Deze kaartlaag staat toe om een WMS laag aan te maken.
+ *
+ * @extends VlMapLayer
+ *
+ * @property {string} data-vl-name - Attribuut bepaalt de kaartlaag naam.
+ * @property {string} data-vl-url - Attribuut bepaalt de WMS url. Verplicht.
+ * @property {string} data-vl-layers - Attribuut bepaalt de layers van de WMS. Verplicht.
+ * @property {string} [data-vl-styles=] - Attribuut bepaalt de WMS stijlen.
+ * @property {string} [data-vl-version=1.3.0] - Attribuut bepaalt de WMS versie.
+ * @property {number} [data-vl-opacity=1] - Attribuut bepaalt de WMS transparantie.
+ * @property {boolean} [data-vl-tiled=true] - Attribuut bepaalt of de WMS tiled wordt opgehaald.
+ *
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-map/releases/latest|Release notes}
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-map/issues|Issues}
+ * @see {@link https://webcomponenten.omgeving.vlaanderen.be/demo/vl-map-wms-layer.html|Demo}
+ */
+export class VlMapWmsLayer extends VlMapLayer {
+  get _url() {
+    return this.getAttribute('data-vl-url') || console.error('url not defined');
+  }
+
+  get _layers() {
+    return this.getAttribute('data-vl-layers') || console.error('layers not defined');
+  }
+
+  get _styles() {
+    return this.getAttribute('data-vl-styles') || '';
+  }
+
+  get _version() {
+    return this.getAttribute('data-vl-version') || '1.3.0';
+  }
+
+  get _opacity() {
+    return Number(this.getAttribute('data-vl-opacity') || 1);
+  }
+
+  get _tiled() {
+    return this.getAttribute('data-vl-tiled') !== 'false';
+  }
+
+  constructor() {
+    super();
+    this._source = this.__createSource();
+    this._layer = this.__createLayer();
+  }
+
+  async connectedCallback() {
+    await super.connectedCallback();
+  }
+
+  __createLayer() {
+    const layerConfig = {
+      title: this._name,
+      source: this._source,
+      opacity: this._opacity,
+      minResolution: this._minResolution,
+      maxResolution: this._maxResolution,
+    };
+    let layer;
+    if (this._tiled) {
+      layer = new OlTileLayer(layerConfig);
+    } else {
+      layer = new OlImageLayer(layerConfig);
+    }
+    layer.set('id', VlMapLayer._counter);
+    return layer;
+  }
+
+  __createSource() {
+    const sourceConfig = {
+      url: this._url,
+      params: {
+        'LAYERS': this._layers,
+        'STYLES': this._styles,
+        'VERSION': this._version,
+      },
+    };
+    if (this._tiled) {
+      return new OlTileWMSSource(sourceConfig);
+    } else {
+      return new OlImageWMSSource(sourceConfig);
+    }
```

Comment:
misschien de if.else structuur tot 1x beperken?

Reply:
in createLayer en createSource that is

Reply:
Ik zou durven zeggen dat dit een smell is en dat het onderscheid moet gemaakt worden door 2 aparte klassen te voorzien hiervoor.

Reply:
Goh, dat is misschien wat overkill hier. Je kan gewoon getLayer doen en daarin je sources inlinen en dan achteraf layer.getSource() doen denk ik.

Reply:
Het zijn 2 aparte soorten source en layer, elk met hun eigen attributen. Als je meer attributen dan de huidige gedeelde attributen wilt laten configureren, gaat dat een zootje worden en ga je wel willen dat het opgedeeld was. Op dat moment ga je dan weer een breaking change moeten voorzien.

Reply:
Eventueel in de constructor zou ik het onderscheid al kunnen maken en dan zo eerst specifieker gaan voor bvb __createTiledLayer,  __createTiledSource, __createSingleImageSource en __createSingleImageLayer of zo iets?

Reply:
Ah ja, klopt in dat opzicht @SanderKleykens. Het idee was om een component te maken per combinatie layer-source zeker ...

Reply:
Consensus is dan maar refactoren in 2 klassen? Eventueel nog een WMS base class die wat gedeelde dingen verzamelt?

Reply:
Nee, akkoord na nog eens opnieuw bekeken te hebben lijkt het wel dat je gelijk hebt. Tiled specifieke attributen zouden hier niet thuishoren in de toekomst dus we splitsen nu beter op. Ik pas het aan.